zend_abstract_interface/functions/php8/functions.c

```diff
@@ -0,0 +1,139 @@
+#include "../functions.h"
+
+#include <assert.h>
+#include <sandbox/sandbox.h>
+
+#ifndef NDEBUG
+static bool z_is_lower(const char *str) {
+    char *p = (char *)str;
+    while (*p) {
+        if (isalpha(*p) && !islower(*p)) return false;
+        p++;
+    }
+    return true;
+}
+#endif
```

Comment:
I think this is missing an include for `<ctype.h>` when it's a debug build.

Reply:
Nice catch. Since this is a pretty common use case in the Zend Engine, and it is [already being used for methods & class lookups](https://github.com/DataDog/dd-trace-php/blob/f4b5065/zend_abstract_interface/methods/php5/methods.c#L7-L16) as well, I was thinking of extracting this into `zai_assert/zai_assert.h` as something like this:

```c
zai_assert_is_lower(str, "String must be lowercase.");
```

Do you think it's worth adding that as part of this PR?

Reply:
Sounds good to me; let's do it.

Reply:
Added in 4e194c58.